File: app/src/main/java/crypto/msd117c/com/cryptocurrency/data/model/list/Coin.java

package crypto.msd117c.com.cryptocurrency.data.model.list;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Coin {

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("symbol")
    @Expose
    private String symbol;
    @SerializedName("slug")
    @Expose
    private String slug;
    @SerializedName("num_market_pairs")
    @Expose
    private Integer numMarketPairs;
    @SerializedName("date_added")
    @Expose
    private String dateAdded;
    @SerializedName("tags")
    @Expose
    private List<String> tags = null;
    @SerializedName("max_supply")
    @Expose
    private Object maxSupply;
    @SerializedName("circulating_supply")
    @Expose
    private Double circulatingSupply;
    @SerializedName("total_supply")
    @Expose
    private Double totalSupply;
    @SerializedName("platform")
    @Expose
    private Object platform;
    @SerializedName("cmc_rank")
    @Expose
    private Integer cmcRank;
    @SerializedName("last_updated")
    @Expose
    private String lastUpdated;
    @SerializedName("quote")
    @Expose
    private Quote quote;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(String slug) {
        this.slug = slug;
    }

    public Integer getNumMarketPairs() {
        return numMarketPairs;
    }

    public void setNumMarketPairs(Integer numMarketPairs) {
        this.numMarketPairs = numMarketPairs;
    }

    public String getDateAdded() {
        return dateAdded;
    }

    public void setDateAdded(String dateAdded) {
        this.dateAdded = dateAdded;
    }

    public List<String> getTags() {
        return tags;
    }

    public void setTags(List<String> tags) {
        this.tags = tags;
    }

    public Object getMaxSupply() {
        return maxSupply;
    }

    public void setMaxSupply(Object maxSupply) {
        this.maxSupply = maxSupply;
    }

    public Double getCirculatingSupply() {
        return circulatingSupply;
    }

    public void setCirculatingSupply(Double circulatingSupply) {
        this.circulatingSupply = circulatingSupply;
    }

    public Double getTotalSupply() {
        return totalSupply;
    }

    public void setTotalSupply(Double totalSupply) {
        this.totalSupply = totalSupply;
    }

    public Object getPlatform() {
        return platform;
    }

    public void setPlatform(Object platform) {
        this.platform = platform;
    }

    public Integer getCmcRank() {
        return cmcRank;
    }

    public void setCmcRank(Integer cmcRank) {
        this.cmcRank = cmcRank;
    }

    public String getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(String lastUpdated) {
        this.lastUpdated = lastUpdated;
    }

    public Quote getQuote() {
        return quote;
    }

    public void setQuote(Quote quote) {
        this.quote = quote;
    }

}

File: app/src/main/java/crypto/msd117c/com/cryptocurrency/domain/coins/model/Coin.kt
package crypto.msd117c.com.cryptocurrency.domain.coins.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Coin(
    @SerializedName("id")
    @Expose
    var id: String? = null,
    @SerializedName("name")
    @Expose
    var name: String? = null,
    @SerializedName("symbol")
    @Expose
    var symbol: String? = null,
    @SerializedName("rank")
    @Expose
    var rank: String? = null,
    @SerializedName("price_usd")
    @Expose
    var priceUsd: String? = null,
    @SerializedName("price_btc")
    @Expose
    var priceBtc: String? = null,
    @SerializedName("24h_volume_usd")
    @Expose
    var _24hVolumeUsd: String? = null,
    @SerializedName("market_cap_usd")
    @Expose
    var marketCapUsd: String? = null,
    @SerializedName("available_supply")
    @Expose
    var availableSupply: String? = null,
    @SerializedName("total_supply")
    @Expose
    var totalSupply: String? = null,
    @SerializedName("max_supply")
    @Expose
    var maxSupply: String? = null,
    @SerializedName("percent_change_1h")
    @Expose
    var percentChange1h: String? = null,
    @SerializedName("percent_change_24h")
    @Expose
    var percentChange24h: String? = null,
    @SerializedName("percent_change_7d")
    @Expose
    var percentChange7d: String? = null,
    @SerializedName("last_updated")
    @Expose
    var lastUpdated: String? = null
)